### DISCUSSION TITLE
Repeated CPU usage increase power consumption

### DISCUSSION THREAD

**Discussion by gvelim** [photoprism/photoprism#2176] [General]
Hi,

On latest photoprism release (02MAR) running on truenas / freebsd (4 core/8 threads), I have observed the following pattern, 5' idle followed by 10' of cpu usage. There are no scans or any other activity reported in the logs.

Is there a way to find out what is causing it and potentially disable it as it increases power consumption on the server running 24/7.

![image](https://user-images.githubusercontent.com/3185416/159488271-b51ac41b-d841-4261-a89a-d1a39e5db50b.png)

**Comment by lastzero** [comment]
I'd assume it's the index background worker updating photo metadata like titles:

https://docs.photoprism.app/getting-started/config-options/#index-workers

You can enable debug/trace logs to see more details.

**Comment by gvelim** [comment]
It appears to be facial recognition based on the below debug logs and matches with the CPU plateau / time window. 

Is this expected, given the library hasn't changed in the last 2 months ?

```
>> Mar 22 13:52:17 photoprism photoprism[38476]: time="2022-03-22T13:52:17Z" level=debug msg="metadata: running facial recognition"
Mar 22 13:52:18 photoprism photoprism[38476]: time="2022-03-22T13:52:18Z" level=debug msg="faces: found no orphan markers [193.848313ms]"
Mar 22 13:52:18 photoprism photoprism[38476]: time="2022-03-22T13:52:18Z" level=debug msg="markers: found no invalid references [201.574641ms]"
Mar 22 13:52:18 photoprism photoprism[38476]: time="2022-03-22T13:52:18Z" level=debug msg="markers: found no missing subjects [892.904µs]"
Mar 22 13:57:54 photoprism photoprism[38476]: time="2022-03-22T13:57:54Z" level=debug msg="http: GET /api/v1/config (200) [207.322572ms]"
Mar 22 14:01:17 photoprism photoprism[38476]: time="2022-03-22T14:01:17Z" level=debug msg="faces: found no ambiguous subjects [8m59.527260654s]"
Mar 22 14:01:18 photoprism photoprism[38476]: time="2022-03-22T14:01:18Z" level=debug msg="faces: found no clusters to be merged [400.445314ms]"
Mar 22 14:01:18 photoprism photoprism[38476]: time="2022-03-22T14:01:18Z" level=debug msg="faces: skipped clustering"
Mar 22 14:01:18 photoprism photoprism[38476]: time="2022-03-22T14:01:18Z" level=debug msg="faces: found no new faces [106.888317ms]"
Mar 22 14:01:18 photoprism photoprism[38476]: time="2022-03-22T14:01:18Z" level=debug msg="faces: found no unmatched markers"
Mar 22 14:01:19 photoprism photoprism[38476]: time="2022-03-22T14:01:19Z" level=debug msg="faces: updated 0 markers, recognized 0 faces, 0 unknown [1.20467048s]"
Mar 22 14:01:19 photoprism photoprism[38476]: time="2022-03-22T14:01:19Z" level=debug msg="metadata: starting routine check"
Mar 22 14:01:20 photoprism photoprism[38476]: time="2022-03-22T14:01:20Z" level=debug msg="moments: analyzing 16784 photos and 1206 videos, with threshold 15"
Mar 22 14:01:21 photoprism photoprism[38476]: time="2022-03-22T14:01:21Z" level=debug msg="moments: 647 albums saved as yaml files"
Mar 22 14:01:21 photoprism photoprism[38476]: time="2022-03-22T14:01:21Z" level=debug msg="index: updating counts"
Mar 22 14:01:21 photoprism photoprism[38476]: time="2022-03-22T14:01:21Z" level=debug msg="counts: updated 0 places [67.691169ms]"
Mar 22 14:01:21 photoprism photoprism[38476]: time="2022-03-22T14:01:21Z" level=debug msg="counts: updated 0 subjects [44.390109ms]"
Mar 22 14:01:21 photoprism photoprism[38476]: time="2022-03-22T14:01:21Z" level=debug msg="counts: updated 0 labels [90.466391ms]"
Mar 22 14:01:21 photoprism photoprism[38476]: time="2022-03-22T14:01:21Z" level=debug msg="index: updating covers"
Mar 22 14:01:21 photoprism photoprism[38476]: time="2022-03-22T14:01:21Z" level=debug msg="covers: updated 0 albums [19.476067ms]"
Mar 22 14:01:21 photoprism photoprism[38476]: time="2022-03-22T14:01:21Z" level=debug msg="covers: updated 0 folders [73.002305ms]"
Mar 22 14:01:21 photoprism photoprism[38476]: time="2022-03-22T14:01:21Z" level=debug msg="covers: updated 0 months [35.737076ms]"
Mar 22 14:01:21 photoprism photoprism[38476]: time="2022-03-22T14:01:21Z" level=debug msg="covers: updated 0 labels [112.159858ms]"
Mar 22 14:01:21 photoprism photoprism[38476]: time="2022-03-22T14:01:21Z" level=debug msg="covers: updated 0 subjects [21.63948ms]"
>> Mar 22 14:07:17 photoprism photoprism[38476]: time="2022-03-22T14:07:17Z" level=debug msg="metadata: running facial recognition"
Mar 22 14:07:18 photoprism photoprism[38476]: time="2022-03-22T14:07:18Z" level=debug msg="faces: found no orphan markers [201.914573ms]"
Mar 22 14:07:18 photoprism photoprism[38476]: time="2022-03-22T14:07:18Z" level=debug msg="markers: found no invalid references [359.656881ms]"
Mar 22 14:07:18 photoprism photoprism[38476]: time="2022-03-22T14:07:18Z" level=debug msg="markers: found no missing subjects [862.657µs]"
Mar 22 14:07:54 photoprism photoprism[38476]: time="2022-03-22T14:07:54Z" level=debug msg="http: GET /api/v1/config (200) [206.562103ms]"
```

**Reply by lastzero** [reply]
Try disabling YAML files and see if things improve.

**Comment by lastzero** [comment]
It's not just facial recognition if you look closely:

```
Mar 22 14:01:20 photoprism photoprism[38476]: time="2022-03-22T14:01:20Z" level=debug msg="moments: analyzing 16784 photos and 1206 videos, with threshold 15"

Mar 22 14:01:21 photoprism photoprism[38476]: time="2022-03-22T14:01:21Z" level=debug msg="moments: 647 albums saved as yaml files"

Mar 22 14:01:21 photoprism photoprism[38476]: time="2022-03-22T14:01:21Z" level=debug msg="index: updating counts"
```

**Reply by gvelim** [reply]
I saw the `647 albums` however I discounted the info as (a) `starting routine check` precedes it, (b) the `16784 photos and 1206 videos` data set hasn't changed for quite some time now, hence I inferred "face recognition" to be the culprit especially when the below logs coincide with CPU load falling.

```
Mar 22 14:01:18 photoprism photoprism[38476]: time="2022-03-22T14:01:18Z" level=debug msg="faces: found no clusters to be merged [400.445314ms]"
Mar 22 14:01:18 photoprism photoprism[38476]: time="2022-03-22T14:01:18Z" level=debug msg="faces: skipped clustering"
```

Is there anything else to investigate or we accept it as standard behaviour ?

**Reply by lastzero** [reply]
Changes can also result from an app version upgrade, updated location information, or manual/external database updates. While it's possible to check for all that given enough time to develop and test this, it is currently not the case. Best to increase the interval as shown in our docs.

**Reply by lastzero** [reply]
See also related GitHub issues and discussions. Related problems and solutions have been discussed quite a few times in the past 👍

**Reply by gvelim** [reply]
Thanks! have set `PHOTOPRISM_WAKEUP_INTERVAL` to `8h`

**Comment by mhbates** [comment]
I have the same/similar question. This background worker takes about 8-10 minutes each time it executes, and then it starts up again shortly thereafter - guessing 15 minute recurrence. Is this normal behavior based on library size (i.e. this is what it will be like all the time), or is it working through some backlog and will eventually catch up and not be as performance-heavy?

**Reply by lastzero** [reply]
It is designed to batch update eg photo titles based on the latest information / metadata. What photos will be updated depends on a timestamp updated by the worker. Some actions may reset the checked time for some or all photos, so these photos get updated in the background next time the worker runs. So it all depends.

**Reply by mhbates** [reply]
Just wanted to report that this pattern of high CPU usage is still happening 2 months later, with PhotoPrism running 24/7. 8-10 minutes of high CPU usage every 15 minutes. Logs similar to what @gvelim posted.

It seems like either PhotoPrism is working through a backlog at a snail's pace, or something is happening to force it to reprocess the entire library on a regular basis.

Is there some documentation of what actions will reset the checked time for all photos, or a way to force it to perform ALL of these checks at one time instead of over several months? I want to keep using PhotoPrism but it is easily the most CPU-hungry application I have on my server.

**Comment by gvelim** [comment]
This feature really gets the electricity bill going 😒

**Reply by lastzero** [reply]
You are welcome to send a pull request with improvements and a full analysis while we are on vacation.

**Comment by centralhardware** [comment]
see https://github.com/photoprism/photoprism/discussions/2520

**Reply by lastzero** [reply]
Increase the worker interval or disable it if you don't need facial recognition while (maybe) somebody is doing an analysis while we are on vacation.

**Comment by rickysarraf** [comment]
I think it is the facial recognition problem your are hitting into.

Maybe run Photoprism with facial support disabled ? Or take a backup of the
db and then reset your faces db entirely. I believe that'll mitigate your
current high cpu usage issue. It will make you lose all the face taggings
though

s3nt fr0m a $martph0ne, excuse typ0s

**Reply by lastzero** [reply]
Possible. Maybe run `photoprism faces audit --fix` first to see if it helps. Some users also reported issues after migrating from Sqlite to MariaDB, see Troubleshooting Guide in our docs. Just saying something is bad or slow and then expecting a solution doesn't work as all we can do is guess.

**Reply by centralhardware** [reply]
IMHO just listen fs changes instead of full rescan a most times hundread of gigabytes avery time

**Reply by lastzero** [reply]
That has nothing to do with the background worker unless you've set up regular reindexing. In that case, regular activity really should not surprise you.

**Reply by lastzero** [reply]
IMHO you can simply use WebDAV for auto indexing functionality or contribute if it bothers you.

**Reply by centralhardware** [reply]
main idea is not autoindex, main idea do not index until any file changed

**Reply by lastzero** [reply]
Yes, that's what WebDAV auto indexing does.

**Reply by lastzero** [reply]
For a FS watcher, the first step is to do research and build a proof of concept. I see nobody doing this. Everyone just keeps telling us it's simple and "should" work. That's not how we work here.

**Reply by lastzero** [reply]
Research means creating a matrix with operating system and filesystem combinations incl any limitations like the number of files / folders supported. I'll enjoy my vacation now and wait for someone doing that! Thanks ❤️

**Reply by lastzero** [reply]
Updating the keywords / faces index in the background is still required and has nothing to do with indexing new files, certainly not gigabytes of data involved.

**Reply by graciousgrey** [reply]
Related issue: https://github.com/photoprism/photoprism/issues/2608

**Comment by gvelim** [comment]
I have tried `photoprism faces audit --fix` which has came up with some missing migrations and which has applied.

But still when I revert the setting from 8h down to 1h I get the same CPU pattern as in the first post.

**Reply by lastzero** [reply]
The metadata is updated regularly. So from time to time this is normal. I'm on vacation, but our source code is public if you need details. User configured cron jobs or WebDAV access may trigger activity as well. Sponsors who need help with their personal installation can reach us directly for assistance. Everyone else can contribute by reading and improving the code as needed. Thank you! ❤️

**Comment by agramner** [comment]
On my server with a 6 core cpu photoprism use 100 % of one core for 30 min each time the backround worker runs and it seems to get worse for every added photo (face).

I've investigated this a bit before and it seems to be the ResolveFaceCollisions function that consume all the cpu power. Looks like it compares all faces against each other and has O(n^2) compexity. It also has been discussed before: #1872.

**Reply by lastzero** [reply]
To be precise, it was discussed that this should not usually happen in practice unless there is some kind of problem.

**Have you enabled debug mode and checked to see if there is anything helpful in the logs?**

For example, a user modified the database tables directly, resulting in a higher number of entries than the implementation was designed for and tested against. From what I know, an external contributor is currently trying to optimize the algorithm.

**Reply by agramner** [reply]
Ok, the only thing I can think of is that I've tagged a lot of faces manually. For example, if I add a new picture of my daughter (kid) photoprism will often detect her face but will almost never tag it automatically. Then, I'll do it manually instead because I would like to be able to find all the pictures with her face in later. Cool if someone is working on optimizing the algorithm!

I do have debug log enabled and this is a part of the log. The bold line marks the 30 min time.

time="2022-08-13T17:28:13Z" level=debug msg="moments: 1779 albums saved as yaml files"
time="2022-08-13T17:28:13Z" level=debug msg="index: updating counts"
time="2022-08-13T17:28:14Z" level=debug msg="counts: updated 0 places [49.743283ms]"
time="2022-08-13T17:28:14Z" level=debug msg="counts: updated 0 subjects [76.806183ms]"
time="2022-08-13T17:28:14Z" level=debug msg="counts: updated 0 labels [125.105617ms]"
time="2022-08-13T17:28:14Z" level=debug msg="index: updating covers"
time="2022-08-13T17:28:14Z" level=debug msg="covers: updated 0 albums [15.704348ms]"
time="2022-08-13T17:28:14Z" level=debug msg="covers: updated 0 folders [71.895036ms]"
time="2022-08-13T17:28:14Z" level=debug msg="covers: updated 0 months [28.483401ms]"
time="2022-08-13T17:28:14Z" level=debug msg="covers: updated 0 labels [105.71066ms]"
time="2022-08-13T17:28:14Z" level=debug msg="covers: updated 0 subjects [35.551411ms]"
time="2022-08-13T17:57:29Z" level=debug msg="metadata: running face recognition"
time="2022-08-13T17:57:29Z" level=debug msg="faces: found no orphan markers [561.938564ms]"
time="2022-08-13T17:57:31Z" level=debug msg="markers: found no invalid references [2.070024223s]"
time="2022-08-13T17:57:31Z" level=debug msg="markers: found no missing subjects [611.1µs]"
**time="2022-08-13T18:28:04Z" level=debug msg="faces: found no ambiguous subjects [30m32.903092057s]"**
time="2022-08-13T18:28:04Z" level=debug msg="faces: found no clusters to be merged [19.416182ms]"
time="2022-08-13T18:28:04Z" level=debug msg="faces: found 153 unclustered samples"
time="2022-08-13T18:28:04Z" level=info msg="faces: found 1 new cluster"
time="2022-08-13T18:28:04Z" level=info msg="faces: skipped cluster CHKFDG6TOOLU5JN4ZO7LNI6IIJIDZSMU, embedding not distinct enough"
time="2022-08-13T18:28:04Z" level=debug msg="faces: found no new faces [334.950611ms]"
time="2022-08-13T18:28:04Z" level=debug msg="faces: found no unmatched markers"
time="2022-08-13T18:28:10Z" level=debug msg="faces: updated 0 markers, recognized 0 faces, 0 unknown [5.39015225s]"

**Reply by lastzero** [reply]
Could it be a problem with the database schema? Did all migrations run? Did you change anything on the database? What database version is it? 

👉 https://docs.photoprism.app/getting-started/advanced/migrations/

**Reply by agramner** [reply]
Yes all the migrations have run successfully. It's MariaDB 10.6 docker image. But it's migrated from the initial SQlite DB

I'm currently backing up the database and performing a full re-index of faces (photoprism faces reset -f && photoprism faces index) to experiment a bit.

**Reply by lastzero** [reply]
👉 https://docs.photoprism.app/getting-started/advanced/migrations/sqlite-to-mariadb/#solving-performance-issues

👉 https://docs.photoprism.app/developer-guide/database/

Next time please provide such important information in advance 👍

**Reply by lastzero** [reply]
I've again improved our docs regarding this, hope it helps as we are not available for help next week: https://github.com/photoprism/photoprism-docs/commit/0f6d81f70de611448101d0206213d30462dabaeb

**Reply by mhbates** [reply]
@agramner Any luck with your experiment? I started Photoprism with MariaDB, no migration, and have the same issue.

Here are my logs showing the issue, with much of the time spent on ambiguous subjects. And I too have manually tagged many faces of my child. I wonder if there's a connection there.

time="2022-08-16T14:22:08Z" level=debug msg="metadata: running face recognition"
time="2022-08-16T14:22:08Z" level=debug msg="faces: found no orphan markers [40.151892ms]"
time="2022-08-16T14:22:08Z" level=debug msg="markers: found no invalid references [94.083347ms]"
time="2022-08-16T14:22:08Z" level=debug msg="markers: found no missing subjects [436.368µs]"
**time="2022-08-16T14:42:14Z" level=debug msg="faces: found no ambiguous subjects [20m5.647909626s]"**
time="2022-08-16T14:42:14Z" level=debug msg="faces: found no clusters to be merged [283.276971ms]"
time="2022-08-16T14:42:14Z" level=debug msg="faces: skipped clustering"
time="2022-08-16T14:42:14Z" level=debug msg="faces: found no new faces [7.909991ms]"
time="2022-08-16T14:42:14Z" level=debug msg="faces: found no unmatched markers"
time="2022-08-16T14:42:15Z" level=debug msg="faces: updated 0 markers, recognized 0 faces, 0 unknown [497.205315ms]"
time="2022-08-16T14:42:15Z" level=debug msg="metadata: starting routine check"
time="2022-08-16T14:42:15Z" level=debug msg="moments: analyzing 24947 photos and 4370 videos, with threshold 15"
time="2022-08-16T14:42:15Z" level=debug msg="moments: 301 albums saved as yaml files"
time="2022-08-16T14:42:15Z" level=debug msg="index: updating counts"
time="2022-08-16T14:42:15Z" level=debug msg="counts: updated 0 places [27.453692ms]"
time="2022-08-16T14:42:15Z" level=debug msg="counts: updated 0 subjects [42.673047ms]"
time="2022-08-16T14:42:15Z" level=debug msg="counts: updated 0 labels [87.818363ms]"
time="2022-08-16T14:42:15Z" level=debug msg="index: updating covers"
time="2022-08-16T14:42:15Z" level=debug msg="covers: updated 0 albums [684.441µs]"
time="2022-08-16T14:42:15Z" level=debug msg="covers: updated 0 folders [42.414497ms]"
time="2022-08-16T14:42:15Z" level=debug msg="covers: updated 0 months [17.190713ms]"
time="2022-08-16T14:42:15Z" level=debug msg="covers: updated 0 labels [93.09902ms]"
time="2022-08-16T14:42:15Z" level=debug msg="covers: updated 0 subjects [19.718176ms]"

**Reply by agramner** [reply]
@mhbates I did a (pretty time consuming) experiment where I started everything from scratch and indexed all my photos again, completely new docker volumes for photoprism and mariadb. I started to do some manual tagging and I already start to see increased CPU usage. So it doesn't seem like there was any problem with my old database (which was migrated from SQLite).

As a side note, I use really fast NVME SSD for storage.

**Reply by mhbates** [reply]
Thanks for the update. Hopefully this helps the devs narrow down the problem. I'm also using an SSD for storage and database, so I don't think I/O is the issue.

**Reply by lastzero** [reply]
We don't see this and have thousands of faces tagged for every person. High CPU usage when tagging new faces is normal. That's not what this discussion is about.

**Reply by mhbates** [reply]
I think what we're saying is that having manually tagged faces seems to come with a long-term CPU performance impact with these background tasks. We're not referring to the immediate CPU impact during the process of tagging.

**Reply by lastzero** [reply]
It should not, but of course we can test this once again after our vacation.

**Comment by lastzero** [comment]
### Why does PhotoPrism always consume 100% of CPU when the background worker is running?

Many users reporting poor performance and high CPU load have migrated from SQLite to MariaDB so that [their database schema is not optimized for performance](https://docs.photoprism.app/getting-started/advanced/databases/), for example, because indexes are missing or columns have the wrong data type. The [instructions for these migrations](https://docs.photoprism.app/getting-started/advanced/migrations/sqlite-to-mariadb/) were provided by a contributor and are not part of the original software distribution. As such, they have not been officially released, recommended, or extensively tested by us.

In some instances, users have manually changed the contents of the database. It is also possible that the database is in an inconsistent state for other reasons, e.g. due to bugs in previous versions that have been fixed in the meantime. However, we are not currently aware of any such cases.

Due to the amount of time required to review each report, we can only offer this to [eligible sponsors](https://photoprism.app/membership) and [business customers](https://photoprism.app/teams), and not to users who have chosen our free community edition.

**Reply by rickysarraf** [reply]
@lastzero I don't think this information may be correct. I started with PhotoPrism only as a MySQL user. With the Facial Recognition feature, I've been bitten with the bug multiple times. With every iteration, I have avoided the practices that have been recommended, like `No many manual face tagging, fiddling with the database` etc.

But the problem still occurs. This is a bug. Maybe a difficult and time consuming one, which will need more resources and time to root cause and come up with a fix. The reason I call it a bug is because once I disabled `Facial Tagging`, I have not run into the problem so far, in almost 6 weeks.

**Reply by lastzero** [reply]
Not everyone has the exact same problem and nobody said the software is completely bug free. There were at least two instances where queries were extremely slow after migrating the database. Queries that should not be slow. Why not inform about this? 

In other cases, we require issues to be reroducable as we can't fix them otherwise. From what I see, we don't have this information from you yet.

**Comment by l8gravely** [comment]
I have around 70,000+ images in my setup, and I ran into a previous problem where I had two seperate users matched to the same name, which was confusing the system.  Since lots of these pictures are of my kids over the years, I've slowly started to tag them as:  "Name, YYYY" in the hopes that this will help the algorithm sort faces better.

This might actually be a good heuristic for future facial recognition, where you take into account the year in the photo meta-data.  This won't be perfect, nothing is, but it might fix things.  

I've just recently upgraded my photoprism server to a quad core AMD Phenom II X4 with 16gb of RAM.  Before I had a quad core intel with only 8gb of RAM and that wasn't enough, the system swapped like crazy since it was also my main desktop.  Anyway... I haven't had enough time lately to poke at this in detail, but I wanted to chime in with some thoughts.

**Comment by dfyx** [comment]
This is still happening over a year later:

![image](https://github.com/photoprism/photoprism/assets/458055/14983c3b-15da-404c-a437-674de4a16981)
![image](https://github.com/photoprism/photoprism/assets/458055/719b1d5d-90ee-4209-935f-84d08ce4eb4c)

You can see pretty well when I turned photoprism off. The peak in read access and the two smaller CPU peaks are unrelated, a friend had started copying something from a network share just when I turned photoprism off.

Hardware specs:
CPU: Intel N100
RAM: 32 GB DDR4
Storage: 4 HDDs in a zfs pool

Photoprism install: docker stack with web app, mariadb and watchtower, slightly adapted from the reference docker-compose.yml
Library: about 70000 photos, mixed RAW and JPEG

I understand that sometimes background jobs need to be run but that doesn't explain this pattern. I would expect a job to start when the library changes, take a few hours and then be done. Starting a job every 30 minutes that pushes one CPU core to 100% for 15-20 minutes even if the library hasn't changed for weeks is clearly a bug. And one that unnecessarily increases my electricity bill and wears out my drives.

Turning off face recognition isn't an option for me as that's the main reason why I'm using photoprism in the first place. I'm not proficient in Go or I'd invest some time to fix this. Isn't there a way to only run these jobs when there's actually any new data. Processing the same photos for the hundredth time won't give any new results.

**Reply by lastzero** [reply]
This is probably related to glitches in the facial recognition data that some users are experiencing (we have never observed this on any of our own instances, which is why the problem was not resolved beforehand):

- https://github.com/photoprism/photoprism/issues/2806
- https://github.com/photoprism/photoprism/issues/3124
- https://docs.photoprism.app/known-issues/#face-recognition

I assume that it won't take many changes to fix it once the cause is known. Checking the corresponding database tables with SQL might be enough, so you don't need to write any Go. If the data is consistent, the worker should not cause a high CPU load if no changes have been made.

We are grateful for any help in identifying the underlying problem that prevents the number of faces in the database from being reduced - it is probably because for some reason there are still references to the face IDs from markers that could not be updated e.g. due to name conflicts.

**Kindly note, though, that we are currently working on additional authentication options such as 2FA, OAuth and OIDC, because many users are also waiting for these. Once that is done, we can work on other issues like this one, which is already labeled as important.**

This year so far:

- we developed a free, high-resolution vector world map that is now available to all users
- released the initial user management that you can use right now
- took care of a long list of pull requests that were waiting to get merged, e.g. the [Places cluster view](https://github.com/photoprism/photoprism/releases/tag/230923-e59851350) that many users had been waiting for
- provided installation packages for Linux as an alternative to our Docker images
- continued to improve our documentation for you
- and fixed virtually all other issues that users reported to us

**Reply by lastzero** [reply]
@dfyx Also note the following improvements, which reduce background activity in other cases:

- https://github.com/photoprism/photoprism/issues/1736
- https://github.com/photoprism/photoprism/issues/2705

These have already been released. So if you have not yet installed our latest version, updating your instance might help.

Edit: There are actually more such improvements that have been added over time. Since I have to work on other things now, I can't find any more links. You might find them if you check our release notes and/or Github Issues.

**Reply by dfyx** [reply]
Thanks for the hints, I will look into them and report back as soon as I find the time.
